samza-core/src/main/java/org/apache/samza/storage/TransactionalStateTaskRestoreManager.java

```diff
@@ -243,17 +243,6 @@ static StoreActions getStoreActions(
           timeSinceLastCheckpointInMs = System.currentTimeMillis() -
               checkpointedChangelogOffset.getCheckpointId().getMillis();
         }
-      
-        // if the clean.store.start config is set, delete the currentDir, restore from oldest offset to checkpointed
-        if (storageEngine.getStoreProperties().isPersistedToDisk() && new StorageConfig(
-          config).getCleanLoggedStoreDirsOnStart(storeName)) {
-          File currentDir = storageManagerUtil.getTaskStoreDir(nonLoggedStoreBaseDirectory, storeName, taskName, taskMode);
```

Comment:
@prateekm @rmatharu shouldn't we be cleaning from loggedStoreBaseDirectory here (in the non-deleted copy of this code block)?

Reply:
Good catch! I suppose the duplicate code block should for the `loggedStore` and should use the `loggedStoreBaseDirectory`. 
Do you mind updating the patch with the changes?

Reply:
This is what I suspected, just wanted to confirm with another set of eyes. I'll update the patch.